### DISCUSSION TITLE
How to set ablation boundary conditions

### DISCUSSION THREAD

**Discussion by shin-chan-cpu** [idaholab/moose#27608] [Q&A Modules: Solid mechanics]
I would like to ask if there is any case in moose where the ablation problem can be calculated? I want to learn it.

**Comment by GiudGiud** [comment]
Hello

What is ablating material from the system?
There's definitely capabilities in moose to remove elements from a mesh, which could conceivably be used to model ablation

Guillaume

**Reply by shin-chan-cpu** [reply]
What type should be used to implement this?

**Reply by GiudGiud** [reply]
maybe a good start would be for you to examine MALAMUTE, which does additivie manufacturing simulations. There are good ressources for modifying the mesh during a simulation there
https://github.com/idaholab/malamute

**Reply by shin-chan-cpu** [reply]
I also have a small question to ask, if I want to calculate the 3D crack propagation calculation and do not set the predetermined crack, is there any userobject that can be used in the XFEM module of moose now?I would be very grateful if you could help me.

**Reply by GiudGiud** [reply]
I think this was asked recently in the forum. Look for posts comparing XFEM to peridynamics maybe

**Reply by shin-chan-cpu** [reply]
![image](https://github.com/idaholab/moose/assets/146061785/166aae9d-7b59-498a-9b95-36dddc1f4512)
I learned the type of the official website, but I didn't find a suitable example. Does this example on the official website have a corresponding input card or something? I would like to refer to it, thank you.

**Reply by GiudGiud** [reply]
hello
Do you have the link to this picture on the website? It's easy enough to find who contributed it

**Reply by shin-chan-cpu** [reply]
And of course, there it is https://mooseframework.inl.gov/modules/peridynamics/index.html

**Reply by GiudGiud** [reply]
@hchen139 do you know if this picture was generated with MOOSE or with Marmot?

**Reply by hchen139** [reply]
The figure is a simulation result with MOOSE using the peridynamics capabilities.

**Reply by shin-chan-cpu** [reply]
Can I refer to your input card, or can you tell me how to calculate 3D cracks with this module?

**Reply by shin-chan-cpu** [reply]
@GiudGiud  I would like to ask when I calculate the crack, even if I set the threshold very small, my crack cannot be introduced. My input is follow.
```
  [XFEM]
    geometric_cut_userobjects = 'cut_mesh'
    qrule = volfrac
    output_cut_plane = true
  []
  [DomainIntegral]
    integrals = 'Jintegral InteractionIntegralKI InteractionIntegralKII'
    displacements = 'disp_x disp_y'
    crack_front_points_provider = cut_mesh
    2d=true
    number_points_from_provider = 0
    crack_direction_method = CurvedCrackFront
    radius_inner = '0.15'
    radius_outer = '0.45'
    youngs_modulus = 6e5
    poissons_ratio = .15
    #block = 'TGO'
    incremental = true
    used_by_xfem_to_grow_crack = true
  []
  
  [UserObjects]
    [nucleate]
      type = MeshCut2DRankTwoTensorNucleation
      tensor = stress
      #scalar_type = VonMisesStress
      initiate_on_boundary = tgo_bc
      scalar_type = MaxPrincipal
      nucleation_threshold = 1e8
      nucleation_length = 4
      #edge_extension_factor = 2e-5
      nucleate_across_full_element = true
    []
    [cut_mesh]
      type = MeshCut2DFractureUserObject
      mesh_file = make_edge_crack_in.e
      k_critical= 1e8
      growth_increment = 4
      nucleate_uo = nucleate
    []
  []
```

**Reply by GiudGiud** [reply]
@hugary1995
